gate-core/src/test/java/com/netflix/spinnaker/gate/services/PermissionServiceTest.java

```
@@ -0,0 +1,320 @@
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.fail;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+import static org.mockito.Mockito.when;
+
+import com.netflix.spinnaker.fiat.model.Authorization;
+import com.netflix.spinnaker.fiat.model.UserPermission;
+import com.netflix.spinnaker.fiat.model.resources.Application;
+import com.netflix.spinnaker.fiat.model.resources.Permissions;
+import com.netflix.spinnaker.fiat.model.resources.Role;
+import com.netflix.spinnaker.fiat.shared.FiatPermissionEvaluator;
+import com.netflix.spinnaker.fiat.shared.FiatStatus;
+import com.netflix.spinnaker.gate.services.internal.ExtendedFiatService;
+import com.netflix.spinnaker.kork.exceptions.SpinnakerException;
+import com.netflix.spinnaker.security.User;
+import java.io.IOException;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Stream;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+import retrofit.RetrofitError;
+import retrofit.client.Response;
+import retrofit.converter.ConversionException;
+
+public class PermissionServiceTest {
+
+  @Test
+  public void lookupServiceAccountReturnsEmptyOn404() {
+    ExtendedFiatService extendedFiatService = mock(ExtendedFiatService.class);
+    String user = "foo@bar.com";
+
+    when(extendedFiatService.getUserServiceAccounts(user)).thenThrow(httpRetrofitError(404));
+
+    PermissionService subject = new PermissionService(null, extendedFiatService, null, null, null);
+
+    try {
+      var result = subject.lookupServiceAccounts(user);
+      assertNotNull(result);
+      assertTrue(result.isEmpty());
+    } catch (Exception e) {
+      fail("No exception should be thrown");
```

Isn't it cleaner to leave out the try/catch and let the exceptions bubble up?  Then junit fails the test.

---

Corrected now.